### PR TITLE
[P4-1497] Prepare includes

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

P4-1497

### What?

We are expecting includes to come from the query param in the following format:

`?include=gender,location,person.profile`

Which gets parsed by rails/rack as:

```ruby
"include" => [
  "gender",
  "location",
  "person.profile"
]
```

This format is supported by AMS in addition to the current format we're using.

This PR moves our current includes to the same format that comes from the query params to avoid doing things in more than one way and to simplify our setup.

### Why?

To simplify our setup

### CODE REVIEW

File: app/controllers/api/v1/moves_controller.rb [hunk 1]
```diff
@@ -8,7 +8,8 @@ class MovesController < ApiController
       def index
         moves = Moves::Finder.new(filter_params, current_ability, params[:sort] || {}).call
         # Excludes potentially many court hearing documents to reduce the request size. This was requested specifically by the frontend team.
-        paginate moves, include: MoveSerializer::INCLUDED_ATTRIBUTES.dup.except(:court_hearings), fields: MoveSerializer::INCLUDED_FIELDS
+        includes = MoveSerializer::INCLUDED_ATTRIBUTES.dup - %w[court_hearings]
+        paginate moves, include: includes, fields: MoveSerializer::INCLUDED_FIELDS
```
Comment (alexdesi) [inline app/controllers/api/v1/moves_controller.rb:12]: what is `includes` ?

File: app/controllers/api/v1/moves_controller.rb [hunk 2]
```diff
@@ -8,7 +8,8 @@ class MovesController < ApiController
       def index
         moves = Moves::Finder.new(filter_params, current_ability, params[:sort] || {}).call
         # Excludes potentially many court hearing documents to reduce the request size. This was requested specifically by the frontend team.
-        paginate moves, include: MoveSerializer::INCLUDED_ATTRIBUTES.dup.except(:court_hearings), fields: MoveSerializer::INCLUDED_FIELDS
+        includes = MoveSerializer::INCLUDED_ATTRIBUTES.dup - %w[court_hearings]
```
Comment (smoothcontract) [inline app/controllers/api/v1/moves_controller.rb:11]: `MoveSerializer::INCLUDED_ATTRIBUTES` constant is only reference here, so would be cleaner to simply remove `court_hearings` from that constant defined in `MoveSerializer`.

File: app/controllers/api/v1/moves_controller.rb (hunk 1 above)
Comment (smoothcontract) [reply app/controllers/api/v1/moves_controller.rb:12]: See line 11 ☝️

File: app/serializers/move_serializer.rb [hunk 3]
```diff
@@ -25,15 +25,16 @@ class MoveSerializer < ActiveModel::Serializer
...
-    allocation: [],
-  }.freeze
+  INCLUDED_ATTRIBUTES = %w[
+    person.ethnicity
+    person.gender
+    from_location
+    to_location
+    documents
+    prison_transfer_reason
+    court_hearings
```
Comment (smoothcontract) [inline app/serializers/move_serializer.rb:35]: Seems like we don't need `court_hearings`, so could remove here.

Comment (cwrw) [reply app/serializers/move_serializer.rb:35]: we are showing `court_hearings` in the single view endpoint though not the list one

File: app/controllers/api/v1/moves_controller.rb (hunk 1 above)
Comment (alexdesi) [reply app/controllers/api/v1/moves_controller.rb:12]: 😄

File: app/serializers/move_serializer.rb (hunk 3 above)
Comment (smoothcontract) [reply app/serializers/move_serializer.rb:35]: Ah, bollocks. You're right - it's referenced in two places inside `moves_controller.rb`. Harrumph.

File: app/controllers/api/v1/moves_controller.rb (hunk 2 above)
Comment (smoothcontract) [reply app/controllers/api/v1/moves_controller.rb:11]: Ignore this comment 🤦

**Comment by willfish** [comment]
@cwrw What do you consider necessary and sufficient to validate this change?

I'm thinking that a simple test of the nested/non-nested dot-separated include forms in the move serializer will do?

**Comment by cwrw** [comment]
I just double checked, and this work might be out of scope for this ticket, but I will come back to them and fix them up later, as was holding off till sparse fields work comes in, but feel free to merge

**Comment by willfish** [comment]
@alexdesi is going to add specs for this. I've manually run these and am comfortable that the behaviour is the same.

**Pull request merged**